packages/happy-dom/src/query-selector/QuerySelector.ts

```diff
@@ -466,6 +443,44 @@ export default class QuerySelector {
 		return null;
 	}
 
+	/**
+	 * Finds elements that match a query selector on a node in order and updates relevant caches
+	 *
+	 * @param items List of items that will be populated with the matches in order.
+	 * @param node Node to search in
+	 * @param selector Selector to match against
+	 * @param cachedItem Cached item.
+	 
+	 */
+	private static insertMatchesInOrder(
+		items: Element[],
+		node: Element | Document | DocumentFragment,
+		selector: string,
+		cachedItem: ICachedQuerySelectorAllItem | ICachedQuerySelectorItem
+	): void {
+		const groups = SelectorParser.getSelectorGroups(selector);
```

Comment:
My main concerns with this change are the following:
* I'm new to this project so I'm afraid this may have some unexpected consequences due to the complex nature of the caches required to make these perform well
* I'm not sure how much this impacts performance of the library all up, particularly in less complicated query selector scenarios than the one described in the bug.